pulsar-client-cpp/lib/CMakeLists.txt

```diff
@@ -0,0 +1,30 @@
+#
+# Copyright 2016 Yahoo Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+file(GLOB PULSAR_SOURCES *.cc lz4/*.c checksum/*.cc)
+
+set (CMAKE_CXX_FLAGS " ${CMAKE_CXX_FLAGS} -msse4.2 -mpclmul")
+
+add_library(pulsarStatic STATIC ${PULSAR_SOURCES})
+add_library(pulsarShared SHARED ${PULSAR_SOURCES})
+
+set_target_properties(pulsarStatic PROPERTIES OUTPUT_NAME pulsar VERSION $ENV{LIB_VERSION})
```

Comment:
Correct, this will set the version on the compiled so library. I was thinking more about having a macro defined like `-DPULSAR_VERSION="1.16"` in gcc command line. Then using a `version.h` header that captures that and finally send the version to the broker in the connect command